cmd/dep/init.go

```diff
@@ -105,39 +107,50 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	sm.UseDefaultSignalHandling()
 	defer sm.Release()
 
-	pd, err := getProjectData(ctx, pkgT, cpr, sm)
-	if err != nil {
-		return err
-	}
+	// Create empty manifest, lock and projectData, and fill them according to
+	// init operation modes. If gopath flag is set, fetch projectData by searching
+	// through GOPATH, else operate in network mode, solve all the dependencies
+	// over network.
 	m := &dep.Manifest{
-		Dependencies: pd.constraints,
+		Dependencies: make(gps.ProjectConstraints),
 	}
+	l := &dep.Lock{}
+	var pd projectData
 
-	// Make an initial lock from what knowledge we've collected about the
-	// versions on disk
-	l := &dep.Lock{
-		P: make([]gps.LockedProject, 0, len(pd.ondisk)),
-	}
+	if cmd.gopath {
+		pd, err = getProjectData(ctx, pkgT, cpr, sm)
```

Comment:
I thought I left a reply here 🤔 maybe my machine got disconnected and it wasn't saved.

So here again, would be better to create a dedicated issue to discuss how we modify `getProjectData()` 😬. Would be great if you could add some more details about the ideas you have in making it better. The function is huge and a lot of stuff seems to be happening in it. 😅